eng/common/TestResources/New-TestResources.ps1.md

```diff
@@ -1,488 +1,526 @@
-```
-
-### -WhatIf
-Shows what would happen if the cmdlet runs.
-The cmdlet is not run.
-
-```yaml
-Type: SwitchParameter
-Parameter Sets: (All)
-Aliases: wi
-
-Required: False
-Position: Named
-Default value: None
-Accept pipeline input: False
-Accept wildcard characters: False
-```
-
-### -Confirm
-Prompts you for confirmation before running the cmdlet.
-
-```yaml
-Type: SwitchParameter
-Parameter Sets: (All)
-Aliases: cf
-
-Required: False
-Position: Named
-Default value: None
-Accept pipeline input: False
-Accept wildcard characters: False
-```
-
-### CommonParameters
-This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable, -InformationAction, -InformationVariable, -OutVariable, -OutBuffer, -PipelineVariable, -Verbose, -WarningAction, and -WarningVariable. For more information, see [about_CommonParameters](https://go.microsoft.com/fwlink/?LinkID=113216).
-
-## INPUTS
-
-## OUTPUTS
-
-## NOTES
-
-## RELATED LINKS
+---
+external help file: -help.xml
+Module Name:
+online version:
+schema: 2.0.0
+---
+
+# New-TestResources.ps1
+
+## SYNOPSIS
+Deploys live test resources defined for a service directory to Azure.
+
+## SYNTAX
+
+### Default (Default)
+```
+New-TestResources.ps1 [-BaseName <String>] [-ResourceGroupName <String>] [-ServiceDirectory] <String>
+ [-TestApplicationId <String>] [-TestApplicationSecret <String>] [-TestApplicationOid <String>]
+ [-SubscriptionId <String>] [-DeleteAfterHours <Int32>] [-DoNotApplyDeleteAfterTag] [-Location <String>]
+ [-Environment <String>] [-ArmTemplateParameters <Hashtable>] [-AdditionalParameters <Hashtable>]
+ [-EnvironmentVariables <Hashtable>] [-CI] [-Force] [-OutFile] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+### Provisioner
+```
+New-TestResources.ps1 [-BaseName <String>] [-ResourceGroupName <String>] [-ServiceDirectory] <String>
+ [-TestApplicationId <String>] [-TestApplicationSecret <String>] [-TestApplicationOid <String>]
+ -TenantId <String> [-SubscriptionId <String>] -ProvisionerApplicationId <String>
+ -ProvisionerApplicationSecret <String> [-DeleteAfterHours <Int32>] [-DoNotApplyDeleteAfterTag]
+ [-Location <String>] [-Environment <String>] [-ArmTemplateParameters <Hashtable>]
+ [-AdditionalParameters <Hashtable>] [-EnvironmentVariables <Hashtable>] [-CI] [-Force] [-OutFile] [-WhatIf]
+ [-Confirm] [<CommonParameters>]
+```
+
+## DESCRIPTION
+Deploys live test resouces specified in test-resources.json or test-resources.bicep
+files to a new resource group.
+
+This script searches the directory specified in $ServiceDirectory recursively
+for files named test-resources.json or test-resources.bicep.
+All found test-resources.json
+and test-resources.bicep files will be deployed to the test resource group.
+
+If no test-resources.json or test-resources.bicep files are located the script
+exits without making changes to the Azure environment.
+
+A service principal may optionally be passed to $TestApplicationId and $TestApplicationSecret.

```

Comment:
Note to self:  Double check the "Owner" statement.   That seems unusually permissive and it's possible that you misinterpreted the script.